integration/spark-common-cluster-test/src/test/scala/org/apache/spark/sql/common/util/QueryTest.scala

```diff
@@ -34,13 +34,20 @@ import org.apache.spark.sql.{CarbonSession, DataFrame, Row, SQLContext}
 import org.scalatest.Suite
 
 import org.apache.carbondata.core.datastore.impl.FileFactory
+import org.apache.carbondata.core.keygenerator.directdictionary.timestamp.TimeStampGranularityConstants
+import org.apache.carbondata.core.util.CarbonProperties
 
 class QueryTest extends PlanTest with Suite {
 
   val LOGGER = LogServiceFactory.getLogService(this.getClass.getCanonicalName)
 
   val DOLLAR = "$"
 
+  // timezone is fix for timestamp related tests
+  TimeZone.setDefault(TimeZone.getTimeZone("GMT"))
+  CarbonProperties.getInstance().addProperty(
+    TimeStampGranularityConstants.CARBON_CUTOFF_TIMESTAMP,
+    TimeStampGranularityConstants.CARBON_CUTOFF_TIMESTAMP_DEFAULT_VAL)
   // Add Locale setting
```

Comment:
These settings will make sure that if the value of dictionary-encoded-timestamp column is less than `1970-01-01 05:30:00`, it will be treated as a bad record.